qemu/tests/balloon_check.py

```diff
@@ -62,11 +62,20 @@ def memory_check(self, step, ballooned_mem):
         error.context("Check memory status %s" % step, logging.info)
         mmem = self.get_ballooned_memory()
         gmem = self.get_memory_status()
-        guest_ballooned_mem = abs(gmem - self.ori_gmem)
-        if (abs(mmem - self.ori_mem) != ballooned_mem or
-                (abs(guest_ballooned_mem - ballooned_mem) > 100)):
-            self.error_report(step, self.ori_mem - ballooned_mem, mmem, gmem)
-            raise error.TestFail("Balloon test failed %s" % step)
+        if ballooned_mem >= self.ori_mem - 100:
+            timeout = float(self.params.get("login_timeout", 600))
+            try:
+                session = self.vm.wait_for_login(timeout=timeout)
+                if utils_misc.get_win_mon_free_mem(session) > 50:
+                    error.TestFail("Balloon_min test failed %s" % step)
```

Comment:
First, the free memory here is the "free" not "available" memory.
When we set windows guest's balloon to a smaller value, ex,(1,100).  The free memory got by cmd "typeperf "\Memory\Free & Zero Page Last Bytes"" is close to 5M, but minutes later, it will become larger but will not greater than 50M.